goosebit/updater/manager.py

```diff
@@ -133,31 +121,59 @@ async def get_device(self) -> Device:
 
         return self.device
 
-    async def save(self) -> None:
-        await self.device.save()
-
     async def update_fw_version(self, version: str) -> None:
         device = await self.get_device()
         device.fw_version = version
+        await device.save(update_fields=["fw_version"])
 
     async def update_hardware(self, hardware: Hardware) -> None:
         device = await self.get_device()
         device.hardware = hardware
+        await device.save(update_fields=["hardware"])
 
     async def update_device_state(self, state: UpdateStateEnum) -> None:
         device = await self.get_device()
         device.last_state = state
+        await device.save(update_fields=["last_state"])
 
-    async def update_last_seen(self, last_seen: int) -> None:
+    async def update_last_connection(self, last_seen: int, last_ip: str) -> None:
         device = await self.get_device()
         device.last_seen = last_seen
-
-    async def update_last_ip(self, last_ip: str) -> None:
-        device = await self.get_device()
         if ":" in last_ip:
             device.last_ipv6 = last_ip
+            await device.save(update_fields=["last_seen", "last_ipv6"])
         else:
             device.last_ip = last_ip
+            await device.save(update_fields=["last_seen", "last_ip"])
+
+    async def update_update(self, update_mode: UpdateModeEnum, firmware: Firmware):
+        device = await self.get_device()
+        device.assigned_firmware = firmware
+        device.update_mode = update_mode
+        await device.save(update_fields=["assigned_firmware_id", "update_mode"])
```

Comment:
Yes, this is correct as the column is called assigned_firmware_id.

Reworked the API integration tests to do the initial setup and state validation through the /api endpoints which then proves that above code works and changing it to `assigned_firmware` would fail the tests.